flow/scripts/fillcell.tcl

```diff
@@ -4,14 +4,16 @@ if {[expr [file exists $::env(REPORTS_DIR)/congestion.rpt] && \
     in DRC viewer to view congestion"
 }
 
-source $::env(SCRIPTS_DIR)/load.tcl
-load_design 5_2_route.odb 4_cts.sdc
-
 if {[env_var_exists_and_non_empty FILL_CELLS]} {
+  source $::env(SCRIPTS_DIR)/load.tcl
+  load_design 5_2_route.odb 4_cts.sdc
+
   set_propagated_clock [all_clocks]
 
   filler_placement $::env(FILL_CELLS)
   check_placement
-}
 
-write_db $::env(RESULTS_DIR)/5_3_fillcell.odb
+  write_db $::env(RESULTS_DIR)/5_3_fillcell.odb
+} else {
+  shell "cp $::env(RESULTS_DIR)/5_2_route.odb $::env(RESULTS_DIR)/5_3_fillcell.odb"
```

Comment:
This has to be shell and not tcl cp because tcl has second resolution and make has higher resolution of timestamps leading to false negatives in dependency checking...

Reply:
How bizarre